qiita_db/metadata_template/prep_template.py

```diff
@@ -456,19 +426,31 @@ def create_qiime_mapping_file(self, prep_template_fp):
 
         # reading files via pandas
         st = load_template_to_dataframe(sample_template_fp)
-        pt = load_template_to_dataframe(prep_template_fp)
+        pt = self.to_dataframe()
+
         st_sample_names = set(st.index)
         pt_sample_names = set(pt.index)
 
         if not pt_sample_names.issubset(st_sample_names):
             raise ValueError(
-                "Prep template is not a sub set of the sample template, files:"
-                "%s %s - samples: %s" % (sample_template_fp, prep_template_fp,
-                                         str(pt_sample_names-st_sample_names)))
+                "Prep template is not a sub set of the sample template, files"
+                "%s - samples: %s" % (sample_template_fp,
+                                      str(pt_sample_names-st_sample_names)))
 
         mapping = pt.join(st, lsuffix="_prep")
         mapping.rename(columns=rename_cols, inplace=True)
 
+        # We cannot ensure that the QIIME-required columns are present in the
+        # metadata map. However, we have to generate a QIIME-compliant mapping
+        # file. Since the user may need a QIIME mapping file, but not these
+        # QIIME-required columns, we are going to create them here and
+        # populate them with the value XXQIITAXX
+        index = mapping.index
+        placeholder = ['XXQIITAXX'] * len(index)
+        for val in viewvalues(rename_cols):
```

Comment:
Do you think it would make sense to raise a warning in this case? The warning would make a note of the columns that would have place-holders.